src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -231,13 +240,52 @@ private void RunChildren()
 
         private void CreateChildWorkItems()
         {
-            _children = new List<WorkItem>();
+            Children = new List<WorkItem>();
 
             foreach (ITest test in _suite.Tests)
                 if (_childFilter.Pass(test))
-                    _children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                {
+                    if (test.Properties.ContainsKey(PropertyNames.Order))
+                    {
+                        Children.Insert(0, WorkItem.CreateWorkItem(test, _childFilter));
+                        _countOrder++;
+                    }
+                    else
+                    {
+                        Children.Insert(Children.Count, WorkItem.CreateWorkItem(test, _childFilter));
+                    }
+                }
+
+
+            if (_countOrder !=0) SortChildren();
         }
 
+        private class WorkItemOrderComparer : IComparer<WorkItem>
+        {
+            /// <summary>
+            /// Compares two objects and returns a value indicating whether one is less than, equal to, or greater than the other.
+            /// </summary>
+            /// <returns>
+            /// A signed integer that indicates the relative values of <paramref name="x"/> and <paramref name="y"/>, as shown in the following table.Value Meaning Less than zero<paramref name="x"/> is less than <paramref name="y"/>.Zero<paramref name="x"/> equals <paramref name="y"/>.Greater than zero<paramref name="x"/> is greater than <paramref name="y"/>.
+            /// </returns>
+            /// <param name="x">The first object to compare.</param><param name="y">The second object to compare.</param>
+            public int Compare(WorkItem x, WorkItem y)
+            {
+                var xKey = int.MaxValue;
+                var yKey = int.MaxValue;
+                if (x.Test.Properties.ContainsKey(PropertyNames.Order)) int.TryParse(x.Test.Properties[PropertyNames.Order][0].ToString(), out xKey);
+                if (y.Test.Properties.ContainsKey(PropertyNames.Order)) int.TryParse(y.Test.Properties[PropertyNames.Order][0].ToString(), out yKey);
+                return xKey.CompareTo(yKey);
```

Comment:
If `xKey` and `yKey` are both `int.MaxValue`, should we switch to ordering by test name so that people who are using the old alphabetical test order hack will not be broken?